### PR TITLE
OpenAPI updates

### PR DESCRIPTION
**Pull request by SimiHunjan**
Signed-off-by: SimiHunjan <simi.hunjan.k@gmail.com>

**Description**:

Updated content

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 1]
```diff
@@ -2230,9 +2248,28 @@ components:
       schema:
         type: string
         pattern: ^\d{1,10}(.\d{1,9})?$
+    tokenIdPathParam:
+      name: tokenId
+      in: path
+      description: The ID of the token
+      example: 0.0.10 or 10
```
Comment (steven-sheehy) [inline hedera-mirror-rest/api/v1/openapi.yml:2255]: We also support the `realm.num` format for this and others.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 2]
```diff
@@ -1893,46 +1893,47 @@ components:
       name: account.id
       in: query
       description: The ID of the account to return information for
+      example: 0.0.10 or 10
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply hedera-mirror-rest/api/v1/openapi.yml:1896]: second that. And this is actually among the failures reported in the github workflow run.

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 3]
```diff
@@ -2230,9 +2248,28 @@ components:
...
+    tokenIdPathParam:
+      name: tokenId
+      in: path
+      description: The ID of the token
+      example: 0.0.10 or 10
+      required: true
+      schema:
+        type: string
+        $ref: '#/components/schemas/EntityId'
+    tokpicIdPathParam:
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-rest/api/v1/openapi.yml:2260]: Also, where is this used?

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 4]
```diff
@@ -1954,49 +1955,56 @@ components:
       schema:
         type: boolean
         default: true
+    contractIdPathParam:
+      name: contractId
+      description: The ID of the contract to return information for
+      example: 0.0.10 or 10
+      in: path
```
Comment (xin-hedera) [inline hedera-mirror-rest/api/v1/openapi.yml:1962]: need to add `required: true`

github workflow failed:

```
-attribute components.parameters.For path parameter contractIdPathParam the required value should be true
```

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 5]
```diff
@@ -1954,49 +1955,56 @@ components:
...
+      schema:
+        $ref: '#/components/schemas/EntityId'
     contractIdQueryParam:
-      name: contract.id
+      name: contractId
       description: The ID of the contract to return information for
       in: query
       examples:
         entityNumNoOperator:
```
Comment (xin-hedera) [inline hedera-mirror-rest/api/v1/openapi.yml:1970]:
```suggestion
        contractNumNoOperator:
```
applies to every `entityNum` in this examples section

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/api/v1/openapi.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SimiHunjan) [reply hedera-mirror-rest/api/v1/openapi.yml]: removed the places where I example and examples existed

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 6]
```diff
@@ -2159,7 +2176,7 @@ components:
         minimum: 1
         maximum: 9223372036854775807
     serialNumberQueryParam:
-      name: serialnumber
+      name: serial.number
```
[the remark this answers is not in the document]
Comment (SimiHunjan) [reply hedera-mirror-rest/api/v1/openapi.yml:2177]: aw darn I thought there was maybe a patten to follow for query params vs. path params in the naming convention. Reverted back.

File: hedera-mirror-rest/api/v1/openapi.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SimiHunjan) [reply hedera-mirror-rest/api/v1/openapi.yml:2257]: Screen Shot 2022-01-12 at 12 58 11 AM

File: hedera-mirror-rest/api/v1/openapi.yml (hunk 1 above)
Comment (SimiHunjan) [reply hedera-mirror-rest/api/v1/openapi.yml:2252]: updated

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 7]
```diff
@@ -1954,49 +1955,56 @@ components:
...
+    contractIdPathParam:
+      name: contractId
+      description: The ID of the contract to return information for
+      example: 0.0.10 or 10
+      in: path
+      schema:
+        $ref: '#/components/schemas/EntityId'
     contractIdQueryParam:
-      name: contract.id
+      name: contractId
```
[the remark this answers is not in the document]
Comment (SimiHunjan) [reply hedera-mirror-rest/api/v1/openapi.yml]: fixed

**Comment by SimiHunjan** [comment]
reference: https://github.com/hashgraph/hedera-mirror-node/pull/3132

**Pull request closed without merging**